userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1580,6 +1663,130 @@ int32_t scap_bpf_get_stats(struct scap_engine_handle engine, OUT scap_stats* sta
 	return SCAP_SUCCESS;
 }
 
+const struct scap_stats_v2* scap_bpf_get_stats_v2(struct scap_engine_handle engine, uint32_t flags, OUT uint32_t* nstats, OUT int32_t* rc)
+{
+	struct bpf_engine *handle = engine.m_handle;
+	int ret;
+	int fd;
+	int offset = 0; // offset in stats buffer
+	*nstats = 0;
+	uint32_t nstats_allocated = handle->m_nstats;
+	scap_stats_v2* stats = handle->m_stats;
+	if (!stats)
+	{
+		*rc = SCAP_FAILURE;
+		return NULL;
+	}
+
+	if ((flags & PPM_SCAP_STATS_KERNEL_COUNTERS) && (BPF_MAX_KERNEL_COUNTERS_STATS <= nstats_allocated))
```

Comment:
probably i would split this `(BPF_MAX_KERNEL_COUNTERS_STATS <= nstats_allocated)` in a dedicated `if` statement in this way we can terminate immediately and return with a clear error message